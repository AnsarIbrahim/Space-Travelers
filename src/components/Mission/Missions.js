import React from 'react';
import { useSelector } from 'react-redux';
import Mission from './Mission';

export default function Missions() {
  const { missions } = useSelector((store) => store.mission);
  return (
    <div className="mission-container">
      <table>
        <thead className="header">
          <th>Mission</th>
          <th>Thaicom</th>
          <th>Status</th>
        </thead>
        {
    missions.map((mission) => {
      const index = missions.indexOf(mission);
      return (
        <Mission
          missionId={mission.id}
          missionName={mission.name}
          description={mission.description}
          reserved={mission.reserved}
          index={index}
          key={index}
        />
      );
    })

}
      </table>
    </div>
  );
}
