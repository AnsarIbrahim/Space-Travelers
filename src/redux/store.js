import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './missions/missionsSlice';
import rocketReducer from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
    rocket: rocketReducer,
  },
});

export default store;
