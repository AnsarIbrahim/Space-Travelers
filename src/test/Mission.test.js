import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../components/Mission/Missions';
import store from '../redux/store';

describe('Missions page', () => {
  it('renders correctly', () => {
    const Mission = renderer.create(
      <Provider store={store}><Missions /></Provider>,
    )
      .toJSON();
    expect(Mission).toMatchSnapshot();
  });
});
