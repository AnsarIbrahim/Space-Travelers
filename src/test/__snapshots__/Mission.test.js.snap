// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions page renders correctly 1`] = `
<div
  className="mission-container"
>
  <table>
    <thead
      className="header"
    >
      <th>
        Mission
      </th>
      <th>
        Thaicom
      </th>
      <th>
        Status
      </th>
    </thead>
  </table>
</div>
`;
